lib/Sema/TypeCheckProtocol.cpp

```diff
@@ -579,14 +583,46 @@ swift::matchWitness(
                    reqDiffAttr->getParameterIndices() &&
                    witnessDiffAttr->parametersMatch(*reqDiffAttr);
           });
+      bool reqDiffAttrSupersetMatch = llvm::any_of(
+          witnessDiffAttrs, [&](const DifferentiableAttr *witnessDiffAttr) {
+            return witnessDiffAttr->getParameterIndices() &&
+                   reqDiffAttr->getParameterIndices() &&
+                   AutoDiffIndexSubset::get(
+                       ctx, witnessDiffAttr->getParameterIndices()->parameters)
+                     ->isSupersetOf(AutoDiffIndexSubset::get(
+                         ctx, reqDiffAttr->getParameterIndices()->parameters));
+          });
       if (!reqDiffAttrMatch) {
-        if (auto *vdWitness = dyn_cast<VarDecl>(witness))
-          return RequirementMatch(
-              getStandinForAccessor(vdWitness, AccessorKind::Get),
-              MatchKind::DifferentiableConflict, reqDiffAttr);
-        else
-          return RequirementMatch(witness, MatchKind::DifferentiableConflict,
-                                  reqDiffAttr);
+        auto implicitDiffAttr = false;
+        if (reqDiffAttrSupersetMatch) {
+          auto *newAttr = DifferentiableAttr::create(
+              ctx, /*implicit*/ true, reqDiffAttr->AtLoc,
+              reqDiffAttr->getRange(), reqDiffAttr->isLinear(),
+              reqDiffAttr->getParameterIndices(), reqDiffAttr->getJVP(),
+              reqDiffAttr->getVJP(), reqDiffAttr->getRequirements());
+          newAttr->setJVPFunction(reqDiffAttr->getJVPFunction());
```

Comment:
The JVP/VJP should not be copied to the new attribute.

Reply:
I removed them.